ee/desktop/runner/server/server.go

```diff
@@ -151,3 +163,42 @@ func (ms *RunnerServer) isAuthTokenValid(authToken string) bool {
 
 	return false
 }
+
+func (ms *RunnerServer) message(w http.ResponseWriter, r *http.Request) {
+	if r.Body == nil {
+		// TODO: log
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+
+	data := make(map[string]interface{})
```

Comment:
As this has an expected structure, why not use a more clear struct?

Reply:
This is also something users can freely send data too. I think we should do a little sanitization before we send to k2. (K2 can handle whatever, but why make trouble). I think the only sanitization that's obvious is capping the outbound message size. Something like 1k?

Reply:
now using struct